src/main/java/seedu/address/logic/commands/TagCommand.java

```diff
@@ -0,0 +1,87 @@
+package seedu.address.logic.commands;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Address;
+import seedu.address.model.person.EmergencyContact;
+import seedu.address.model.person.Level;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Note;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.person.Subject;
+
+/**
+ * Tags an existing student in the address book with a subject, school level or both.
+ */
+public class TagCommand extends Command {
+
+    public static final String COMMAND_WORD = "tag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + "Tags students";
+
+    public static final String MESSAGE_TAG_PERSON_SUCCESS = "Tagged Student: %1$s";
+
+    private final Name nameToTag;
+    private final EditCommand.EditPersonDescriptor tagsToAdd;
+
+    /**
+     * @param nameToTag name of the person in the address book to tag
+     * @param tagsToAdd details to tag the person with
+     */
+    public TagCommand(Name nameToTag, EditCommand.EditPersonDescriptor tagsToAdd) {
+        this.nameToTag = nameToTag;
+        this.tagsToAdd = tagsToAdd;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        Optional<Person> optionalPersonToTag = lastShownList.stream()
+                .filter(x -> x.getName()
+                        .equals(nameToTag)).findFirst();
+
+
+        Person personToTag;
+        if (optionalPersonToTag.isPresent()) {
+            personToTag = optionalPersonToTag.get();
+        } else {
+            throw new CommandException("Student does not exist in address book");
```

Comment:
Maybe can have it as a Constant so its easier to find and edit the message next time?

Reply:
like same as line 67